examples/pytorch/ogb/ogbn-mag/hetero_rgcn.py

```diff
@@ -0,0 +1,530 @@
+            print(f'   Final Test: {r.mean():.2f} ± {r.std():.2f}')
+
+
+def parse_args():
+    # DGL
+    parser = argparse.ArgumentParser(description='RGCN')
+    parser.add_argument("--dropout", type=float, default=0.5,
+            help="dropout probability")
+    parser.add_argument("--n-hidden", type=int, default=64,
+            help="number of hidden units")
+    parser.add_argument("--lr", type=float, default=0.01,
+            help="learning rate")
+    parser.add_argument("-e", "--n-epochs", type=int, default=3,
+            help="number of training epochs")
+
+    # OGB
+    parser.add_argument('--runs', type=int, default=10)
+
+    args = parser.parse_args()
+    return args
+
+def prepare_data(args):
+    dataset = DglNodePropPredDataset(name="ogbn-mag")
+    split_idx = dataset.get_idx_split()
+    g, labels = dataset[0] # graph: dgl graph object, label: torch tensor of shape (num_nodes, num_tasks)
+    labels = labels['paper'].flatten()
+
+    def add_reverse_hetero(g, combine_like=True):
+        r"""
+        Parameters
+        ----------
+        g : DGLGraph
+            The heterogenous graph where reverse edges should be added
+        combine_like : bool, optional
+            Whether reverse-edges that have identical source/destination 
+            node types should be combined with the existing edge-type, 
+            rather than creating a new edge type.  Default: True.
+        """
+        relations = {}
+        num_nodes_dict = {ntype: g.num_nodes(ntype) for ntype in g.ntypes}
+        for metapath in g.canonical_etypes:
+            src_ntype, rel_type, dst_ntype = metapath
+            src, dst = g.all_edges(etype=rel_type)
+
+            if src_ntype==dst_ntype and combine_like:
+                # Make edges un-directed instead of making a reverse edge type
+                relations[metapath] = (th.cat([src, dst], dim=0), th.cat([dst, src], dim=0))
+            else:
+                # Original edges
+                relations[metapath] = (src, dst)
+
+                reverse_metapath = (metapath[2], 'rev-' + metapath[1], metapath[0])
+                relations[reverse_metapath] = (dst, src)           # Reverse edges
+
+        new_g = dgl.heterograph(relations, num_nodes_dict=num_nodes_dict)
+        # Remove duplicate edges
+        new_g = dgl.to_simple(new_g, return_counts=None, writeback_mapping=False, copy_ndata=True)
+
+        # copy_ndata:
+        for ntype in g.ntypes:
+            for k, v in g.nodes[ntype].data.items():
+                new_g.nodes[ntype].data[k] = v.detach().clone()
+
+        return new_g
+
+    g = add_reverse_hetero(g)
+    print("Loaded graph: {}".format(g))
+
+    logger = Logger(args['runs'], args)
+
+    # train sampler
+    sampler = dgl.dataloading.MultiLayerNeighborSampler(args['fanout'])
+    train_loader = dgl.dataloading.NodeDataLoader(
+        g, split_idx['train'], sampler,
+        batch_size=args['batch_size'], shuffle=True, num_workers=32)
```

Comment:
Using `num_workers=32` will raise an error on g4dn.xlarge, perhaps due to the memory limitation. By setting `num_workers=0`, it runs smoothly with no clear speed difference. I will prefer using a default value of 0 and perhaps you can make it an argument in ArgumentParser.